admin-dev/themes/new-theme/js/pages/product/edit/product-partial-updater.js

```diff
@@ -167,10 +186,57 @@ export default class ProductPartialUpdater {
   getFormDataAsObject() {
     const formArray = this.$productForm.serializeArray();
     const serializedForm = {};
+
     formArray.forEach((formField) => {
-      serializedForm[formField.name] = formField.value;
+      let {value} = formField;
+      if (formField.name.endsWith('[]')) {
+        let multiField = [];
+
+        if (Object.prototype.hasOwnProperty.call(serializedForm, formField.name)) {
+          multiField = serializedForm[formField.name];
+        }
+
+        multiField.push(formField.value);
+        value = multiField;
+      }
+
+      serializedForm[formField.name] = value;
     });
 
     return serializedForm;
   }
+
+  /**
+   * @param $form
+   * @param name
+   * @param value
+   *
+   * @private
+   */
+  appendInputToForm($form, name, value) {
+    $('<input>').attr({
+      name,
+      type: 'hidden',
+      value,
+    }).appendTo($form);
+  }
+
+  // @todo: move out or use some lib for this?
+  arrayEquals(arr1, arr2) {
+    if (arr1.length !== arr2.length) {
+      return false;
+    }
+
+    for (let i = 0; i < arr2.length; i += 1) {
+      if (arr1[i] instanceof Array && arr2[i] instanceof Array) {
+        if (!arr1[i].equals(arr2[i])) {
```

Comment:
```suggestion
        if (!arrayEquals(arr1[i], arr2[i])) {
```
Why not use your method since `equals` was apparently not enough previously?

Besides this means we could have multi dimensional array fields? some fields like `product[carrier][][]` ? I'm not sure it can happen

Reply:
I doubt it, but actually I was thinking to move this method to some util tools or search for some libraries for this. (now its almost just copy paste from some article in stack overflow :smile: )